pkg/infra/process/root_check.go

```diff
@@ -0,0 +1,17 @@
+// +build !windows
+
+package process
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+)
+
+func isServerRunningAsRoot() (bool, error) {
+	serverProcessUser, err := user.Current()
+	if err != nil {
+		return false, fmt.Errorf("could not get current OS user to detect process privileges")
+	}
+	return (serverProcessUser != nil && serverProcessUser.Username == "root") || os.Geteuid() == 0, nil
```

Comment:
We could also verify if running in `production` the user is `grafana` with UID `472` if that makes sense

Reply:
Not sure that would work in all different circumstances? I know that we use that in Docker images, but not sure outside of Docker?

Reply:
I documented what I could [here](https://docs.google.com/document/d/1FiqtJx8XeZ1c-vWBTt7hbVyAXyybFHqzPkQwD_T0XKg/edit#). Let me know if anything is incorrect

Reply:
Just a question: do we want to check if the process is running as _root_ specifically, or as any other (potentially escalated) user? In other words, are we looking for `geteuid() == 0` or `geteuid() != getuid()`?

Reply:
It's a good question - probably both together are good checks. Ideally we would have a pretty consistent standard, and we can just check against that but not 100% yet if I've covered all bases.